ucp/_libs/ucx_api.pyx

```diff
@@ -386,6 +388,82 @@ cdef class UCXWorker(UCXObject):
         assert_ucs_status(status)
         return UCXEndpoint(self, <uintptr_t>ucp_ep)
 
+    cpdef ucs_status_t fence(self) except *:
+        cdef ucs_status_t status = ucp_worker_fence(self._handle)
+        assert_ucs_status(status)
+        return status
+
+    def flush(self, cb_func, cb_args, cb_kwargs=dict()):
+        cdef ucs_status_ptr_t req
+        cdef ucp_send_callback_t _send_cb = <ucp_send_callback_t>_send_callback
+
+        cdef ucs_status_ptr_t status = ucp_worker_flush_nb(self._handle, 0, _send_cb)
+        return _handle_status(
+            status, 0, cb_func, cb_args, cb_kwargs, 'flush', self._inflight_msgs
+        )
+
+    def get_address(self):
+        return UCXAddress(self)
+
+
+def _ucx_address_finalizer(uintptr_t handle_as_int, UCXWorker worker):
+    cdef ucp_address_t *address = <ucp_address_t *>handle_as_int
+    cdef ucp_worker_h ucp_worker = <ucp_worker_h><uintptr_t>worker.handle
+    ucp_worker_release_address(ucp_worker, address)
+
+
+cdef class UCXAddress(UCXObject):
+    """Python representation of ucp_address_t"""
+    cdef ucp_address_t *_handle
+    cdef size_t _length
+    cdef worker
+
+    def __init__(self, UCXWorker worker):
+        cdef ucs_status_t status
+        cdef ucp_worker_h ucp_worker = <ucp_worker_h><uintptr_t>worker.handle
+
+        status = ucp_worker_get_address(ucp_worker, &self._handle, &self._length)
+        assert_ucs_status(status)
+        self.worker = worker
+        self.add_handle_finalizer(
+            _ucx_address_finalizer,
+            int(<uintptr_t>self._handle),
+            worker
+        )
+        worker.add_child(self)
+
+    @property
+    def address(self):
+        return <uintptr_t>self._handle
+
+    @property
+    def length(self):
+        return int(self._length)
+
+    def __getbuffer__(self, Py_buffer *buffer, int flags):
+        if (flags & PyBUF_WRITABLE) == PyBUF_WRITABLE:
+            raise BufferError("Requested writable view on readonly data")
+        buffer.buf = <void*>self._handle
+        buffer.obj = self
+        buffer.len = self._length
+        buffer.readonly = True
+        buffer.itemsize = self._length
+        if (flags & PyBUF_FORMAT) == PyBUF_FORMAT:
+            buffer.format = b"p"
```

Comment:
Should be capitalized if we want `void *`. The lower cased one is `char[]`.

```suggestion
            buffer.format = b"P"
```

https://docs.python.org/3/library/struct.html#format-characters

Reply:
That description sounds to me like an array of (void*)s. The ucp_address_t isn't an array of (void*)s, it's packed bytes. So it seems to me that the char[] is the right type.

Reply:
Ah ok. Sorry if I misunderstood. In that case maybe we should be doing `sizeof(ucp_address_t)` in a few places below and leave this as-is.

Reply:
The trouble is that ucp_address_t is an incomplete type, so using `sizeof(ucp_address_t)` won't work. 
https://github.com/openucx/ucx/blob/8b6641b02ef5d217323f719d54dfb2e9142dad59/src/ucp/api/ucp_def.h#L110

Reply:
Hmm...so how do we know how much memory this takes up?

Reply:
The `ucp_worker_get_address()` function has a length output argument that tells you the length. So you get a byte string and a length and that gets stored in the UCXAddress object.

Reply:
Ah ok. If it is just bytes, then maybe we should just type this as `"B"` (IOW `uint8_t`) and the `itemsize` is just `1`.

Reply:
Ok so if that works, have added a couple suggestions below based on that change ( https://github.com/rapidsai/ucx-py/pull/594#pullrequestreview-500851529 ).